File: backend/app/schemas/brochure.py
from pydantic import BaseModel, Field
from typing import List, Optional, Literal, Dict, Any


class BrochureInput(BaseModel):
    title: Optional[str] = Field(None, description="Brochure title")
    subtitle: Optional[str] = Field(None, description="Optional subtitle")
    address: Optional[str] = Field(None, description="Property address")
    price: Optional[str] = Field(None, description="Price label e.g. $1,250,000")
    bedrooms: Optional[int] = Field(None, ge=0)
    bathrooms: Optional[float] = Field(None, ge=0)
    area_sqft: Optional[int] = Field(None, ge=0)
    property_type: Optional[str] = None
    highlights: Optional[List[str]] = Field(default=None, description="Key highlights")
    amenities: Optional[List[str]] = Field(default=None, description="Amenities list")


class BrochureResult(BaseModel):
    task_id: str
    file_url: str
    status: Literal["completed"]

class BrochureDraftCreate(BaseModel):
    templateKey: str = Field("clean-minimal")
    data: Optional[Dict[str, Any]] = None
    property_id: Optional[str] = None


class BrochureDraftUpdate(BaseModel):
    data: Optional[Dict[str, Any]] = None
    status: Optional[str] = None
    download_url: Optional[str] = None
    error: Optional[str] = None


class BrochureDraftOut(BaseModel):
    id: str
    data: Dict[str, Any]
    status: str
    download_url: Optional[str]
    created_at: str
    updated_at: str


class BrochureTemplateOut(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    file_path: str
    created_at: str

File: aura-client/src/features/brochure/api/brochure.ts
import { api } from '../../../services/http'

const USE_REAL = import.meta.env.VITE_USE_REAL_API === 'true'

export interface BrochureDraftOut {
  id: string
  data: any
  status: 'draft' | 'rendering' | 'ready' | 'error' | string
  download_url?: string | null
  created_at: string
  updated_at: string
}

export interface BrochureTemplateOut {
  id: string
  name: string
  description?: string | null
  file_path: string
  created_at: string
}

export async function createDraft(payload: {
  templateKey: string
  property_id?: string
  data?: any
}): Promise<BrochureDraftOut> {
  if (!USE_REAL) {
    return {
      id: `draft-${Date.now()}`,
      data: { templateKey: payload.templateKey, ...payload.data },
      status: 'draft',
      download_url: null,
      created_at: new Date().toISOString(),
      updated_at: new Date().toISOString(),
    }
  }

  const { data } = await api.post<BrochureDraftOut>('/brochures', payload)
  return data
}

export async function getDraft(id: string): Promise<BrochureDraftOut> {
  const { data } = await api.get<BrochureDraftOut>(`/brochures/${id}`)
  return data
}

export async function updateDraft(
  id: string,
  patch: Partial<BrochureDraftOut>
): Promise<BrochureDraftOut> {
  const { data } = await api.patch<BrochureDraftOut>(`/brochures/${id}`, patch)
  return data
}

export async function renderDraft(id: string): Promise<{ download_url: string }> {
  if (!USE_REAL) {
    await new Promise((resolve) => setTimeout(resolve, 1000))
    return {
      download_url: `http://localhost:8000/api/v1/assets/mock-brochures/${id}.pdf`,
    }
  }

  const { data } = await api.post<{ download_url: string }>(`/brochures/${id}/render`)
  return data
}

export async function getDownloadUrl(id: string): Promise<{ download_url: string }> {
  const { data } = await api.get<{ download_url: string }>(`/brochures/${id}/download`)
  return data
}

export async function listTemplates(): Promise<BrochureTemplateOut[]> {
  if (!USE_REAL) {
    return [
      {
        id: 'mock-clean',
        name: 'Clean Minimal',
        description: 'Light, modern brochure with focus on imagery.',
        file_path: 'templates/brochure/clean-minimal.html',
        created_at: new Date().toISOString(),
      },
      {
        id: 'mock-luxury',
        name: 'Luxury Showcase',
        description: 'Premium layout for high-end properties.',
        file_path: 'templates/brochure/luxury-showcase.html',
        created_at: new Date().toISOString(),
      },
      {
        id: 'mock-neighborhood',
        name: 'Neighborhood Highlight',
        description: 'Area highlights with local amenities.',
        file_path: 'templates/brochure/neighborhood-highlight.html',
        created_at: new Date().toISOString(),
      },
    ]
  }

  const { data } = await api.get<BrochureTemplateOut[]>('/templates')
  return data
}

export async function listDrafts(limit = 20, offset = 0): Promise<BrochureDraftOut[]> {
  if (!USE_REAL) {
    return []
  }

  const params = new URLSearchParams({
    limit: String(limit),
    offset: String(offset),
  })

  const { data } = await api.get<BrochureDraftOut[]>(`/brochures?${params.toString()}`)
  return data
}
